qiskit/transpiler/passes/optimization/template_matching/template_substitution.py

```diff
@@ -375,3 +387,89 @@ def run_dag_opt(self):
 
         self.dag_dep_optimized = dag_dep_opt
         self.dag_optimized = dagdependency_to_dag(dag_dep_opt)
+
+    def _fake_bind(self, template_sublist, circuit_sublist):
+        """
+        Copies the template and fake binds any parameters.
+        template_sublist and circuit_sublist match up to the
+        assignment of the parameters. For example the template
+             ┌───────────┐                  ┌────────┐
+        q_0: ┤ P(-1.0*β) ├──■────────────■──┤0       ├
+             ├───────────┤┌─┴─┐┌──────┐┌─┴─┐│  CZ(β) │
+        q_1: ┤ P(-1.0*β) ├┤ X ├┤ P(β) ├┤ X ├┤1       ├
+             └───────────┘└───┘└──────┘└───┘└────────┘
+        should only match once in the circuit
+             ┌───────┐
+        q_0: ┤ P(-2) ├──■────────────■────────────────────────────
+             ├───────┤┌─┴─┐┌──────┐┌─┴─┐┌──────┐
+        q_1: ┤ P(-2) ├┤ X ├┤ P(2) ├┤ X ├┤ P(3) ├──■────────────■──
+             └┬──────┤└───┘└──────┘└───┘└──────┘┌─┴─┐┌──────┐┌─┴─┐
+        q_2: ─┤ P(3) ├──────────────────────────┤ X ├┤ P(3) ├┤ X ├
+              └──────┘                          └───┘└──────┘└───┘
```

Comment:
I think you want to say "maximally match" here because even in the P(3) case there's still a match of CX-P(3)-CX with the middle part of the template.

Reply:
Yes, I amended the docstring.